keps/sig-auth/4940-psa-block-host-field-in-probes/README.md

```diff
@@ -0,0 +1,354 @@
+probes with the Host field set when using the (about to be deprecated) API.
+This is implemented by [kubernetes PR 125271](https://github.com/kubernetes/kubernetes/pull/125271)
+that does exactly that. Once we add the PSA, we can also deprecate the API
+as part of this KEP itself.
+
+### Risks and Mitigations
+
+There might be users who depend on the `Host` field in
+their existing probes which will continue to work and if
+newly created probes also need the `Host` field to point
+to an external destination then the admin can avoid enforcing
+the PSA to block it.
+
+## Design Details
+
+Add a Baseline APILevel Pod Security Admission policy to allow admins of the
+cluster to block users from setting `.host` field in:
+
+* container.LivenessProbe.ProbeHandler.HTTPGet.Host
+* container.ReadinessProbe.ProbeHandler.HTTPGet.Host
+* container.StartupProbe.ProbeHandler.HTTPGet.Host
+* container.LivenessProbe.ProbeHandler.TCPSocket.Host
+* container.ReadinessProbe.ProbeHandler.TCPSocket.Host
+* container.StartupProbe.ProbeHandler.TCPSocket.Host
+* container.Lifecycle.PostStart.TCPSocket.Host // Deprecated. TCPSocket is NOT supported as a LifecycleHandler and kept for backward compatibility.
+* container.Lifecycle.PreStop.TCPSocket.Host // Deprecated. TCPSocket is NOT supported as a LifecycleHandler and kept for backward compatibility.
+* container.Lifecycle.PostStart.HTTPGet.Host
+* container.Lifecycle.PreStop.HTTPGet.Host
+
+These above exact set of fields is what will get deprecated as well.
+
+### Test Plan
+
+* Unit and E2E tests will be added to ensure the PSA works as expected
+
+##### Prerequisite testing updates
+
+None
+
+##### Unit tests
+
+Necessary unit tests will be added to the [PSA package] for
+testing the new code.
+Current test coverage status for the package is:
+- `k8s.io/pod-security-admission/policy`: `2025-05-06` - `89.9%`
+- `k8s.io/pod-security-admission/test`: `TBD` - `TBD`
+
+[PSA package]: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/pod-security-admission/policy
+
+##### Integration tests
+
+The following integration tests will be added to verify the PSA validation logic:
+
+1. Test that pods with `.host` field set in probes are rejected when PSA is enabled with baseline level
+2. Test that pods without `.host` field set in probes are allowed when PSA is enabled with baseline level
+3. Test that existing pods with `.host` field set continue to work when PSA is enabled
+4. Test that pods with `.host` field set are allowed when PSA is disabled or using an older version
+
+These tests will be added to:
+- `test/integration/auth/podsecurity_test.go`
+https://storage.googleapis.com/k8s-triage/index.html?test=TestPodSecurity
+
+The integration tests will verify the PSA policy validation logic by:
+- Creating test cases for each probe type (HTTPGet, TCPSocket)
+- Testing each probe location (LivenessProbe, ReadinessProbe, StartupProbe, Lifecycle)
```

Comment:
make sure there's a test exercising PSA validation of ephemeral containers when added via the `update pods/ephemeralcontainers` subresource

Reply:
goodpoint! added this also to the test case. I wasn't aware day2 pod can be modified like that. thanks for pointing this out!

Reply:
@liggitt I had almost committed this when I noticed:
```
	// Probes are not allowed for ephemeral containers.
	// +optional
	LivenessProbe *Probe
	// Probes are not allowed for ephemeral containers.
	// +optional
	ReadinessProbe *Probe
	// Probes are not allowed for ephemeral containers.
	// +optional
	StartupProbe *Probe
	// Lifecycle is not allowed for ephemeral containers.
	// +optional
	Lifecycle *Lifecycle
```
so I assume these fields are not applicable to ephemeral containers?